spec/02-integration/05-proxy/01-router_spec.lua

```diff
@@ -499,4 +499,44 @@ describe("Router", function()
         check(i, args[3], args[4]))
     end
   end)
+
+  describe("url encoded uris", function()
+
+    setup(function()
+      helpers.dao:truncate_tables()
+      assert(helpers.dao.apis:insert {
+        name         = "localbin",
+        upstream_url = "http://localhost:9999/",
+        uris         = {
+          "/",
+        },
+        hosts        = {
+          "localbin.com",
+        },
+      })
+
+      assert(helpers.start_kong {
+        nginx_conf = "spec/fixtures/custom_nginx.template",
```

Comment:
One might argue, that we test here `ngx.var.request_uri`. But, I don't mind changing it to `httpbin.org`.